src/storage/hummock_sdk/src/key.rs

```diff
@@ -150,6 +152,49 @@ fn next_key_no_alloc(key: &[u8]) -> Option<(&[u8], u8)> {
 
 // End Copyright 2016 TiKV Project Authors. Licensed under Apache-2.0.
 
+/// Get the end bound of the given `prefix` when transforming it to a key range.
+fn end_bound_of_prefix(prefix: &[u8]) -> Bound<Vec<u8>> {
+    let next_key = next_key(prefix);
+    if next_key.is_empty() {
+        Unbounded
+    } else {
+        Excluded(next_key)
+    }
+}
+
+/// Transform the given `prefix` to a key range.
+pub fn range_of_prefix(prefix: &[u8]) -> impl RangeBounds<Vec<u8>> {
+    if prefix.is_empty() {
+        (Unbounded, Unbounded)
```

Comment:
Is this case possible?

Reply:
Currently nope. Just to make it more robust.

Reply:
What about panic it

Reply:
The hummock storage will panic on an empty key. The prefix can be partial (like encoded pk prefix) in the future so it might be empty.